pkg/controllers/nodetemplate/controller.go

```diff
@@ -94,14 +95,15 @@ func (c *Controller) resolveSubnets(ctx context.Context, nodeTemplate *v1alpha1.
 		nodeTemplate.Status.Subnets = append(nodeTemplate.Status.Subnets, status)
 	}
 
-	if c.client.Status().Update(ctx, nodeTemplate) != nil {
+	if c.client.Status().Patch(ctx, nodeTemplate, client.MergeFrom(stored)) != nil {
```

Comment:
What about if this isn't found? Should probably use `client.IgnoreNotFound` here